### DISCUSSION TITLE
How to exclude a folder ?

### DISCUSSION THREAD

**Discussion by Ange7** [deployphp/deployer#3986] [Help needed]
Hello,

I’m sorry if my question seems "stupid," but I find the documentation for the "Deployer" tool very inconvenient and light. I’m looking for a way to exclude one or more folders during my deployment.

Currently, my configuration looks like this:

```
<?php
namespace Deployer;

require 'recipe/laravel.php';

// Config
set('repository', 'xxxxxxx');

add('shared_files', ['auth.json']);
add('shared_dirs', ['storage']);
add('writable_dirs', ['storage']);

// Hosts
host('xxxx')
    ->set('remote_user', 'user')
    ->set('deploy_path', '/srv/matrix');

task('deploy:copy_dirs');

task('deploy:env', function () {
    upload('.env.production', '{{release_path}}/.env');
    upload('auth.json', '{{release_path}}/auth.json');
    upload('vendor', '{{release_path}}');
});

task('deploy:vendors', function () {
    writeln('Stop it');
});

// Hooks
after('deploy:failed', 'deploy:unlock');
```

I would like tu exclude ".git" folder for example. or ./deployer.phar file. 

Thank you in advance!

**Comment by antonmedv** [comment]
Take a look at this task: https://deployer.org/docs/8.x/recipe/deploy/update_code#update_code_strategy

It uses `git archive` by default. You can use `.gitattributes` to exclude certain files.
